pallets/ah-migrator/src/scheduler.rs

```diff
@@ -136,3 +137,159 @@ impl<T: Config> Pallet<T> {
+		assert!(
+			pallet_rc_migrator::scheduler::alias::Lookup::<T>::iter().next().is_none(),
+			"Lookup map should be empty on asset hub before migration"
+		);
+
+		// Assert storage 'Scheduler::Retries::ah_pre::empty'
+		assert!(
+			pallet_scheduler::Retries::<T>::iter().next().is_none(),
+			"Retries map should be empty on asset hub before migration"
+		);
+	}
+
+	fn post_check(rc_pre_payload: Self::RcPrePayload, _ah_pre_payload: Self::AhPrePayload) {
+		let rc_payload = RcPrePayload::<T>::decode(&mut &rc_pre_payload[..])
+			.expect("Failed to decode RcPrePayload bytes");
+
+		// Assert storage 'Scheduler::IncompleteSince::ah_post::correct'
+		assert_eq!(
+			pallet_scheduler::IncompleteSince::<T>::get(),
+			rc_payload.incomplete_since,
+			"IncompleteSince on Asset Hub should match the RC value"
+		);
+
+		// Mirror the Agenda conversion in `do_process_scheduler_message` above ^ to construct
+		// expected Agendas. Critically, use the passed agenda call encodings to remove reliance
+		// on pallet-preimage state, which will have been changed from the actual migration.
+		let mut expected_ah_agenda: Vec<_> = rc_payload
+			.agenda_and_call_encodings
+			.into_iter()
+			.filter_map(|(block_number, rc_tasks, rc_task_call_encodings)| {
+				let mut ah_tasks = Vec::new();
+				let tasks = rc_tasks.into_iter();
+				let encoded_calls = rc_task_call_encodings.into_iter();
+
+				// Iterate over task and it's corresponding encoded call.
+				for (rc_task_opt, encoded_call_opt) in tasks.zip(encoded_calls) {
+					// Check both task and encoded call exist.
+					if let (Some(rc_task), Some(encoded_call)) = (rc_task_opt, encoded_call_opt) {
+						// Attempt origin conversion.
+						let ah_origin =
+							match T::RcToAhPalletsOrigin::try_convert(rc_task.origin.clone()) {
+								Ok(origin) => origin,
+								Err(_) => {
+									// Origin conversion failed, skip task.
+									continue;
+								},
+							};
+
+						// Attempt call conversion.
+						let ah_call = match Pallet::<T>::map_rc_ah_call_no_preimage(encoded_call) {
+							Ok(call) => call,
+							// Conversion failed, skip task.
+							Err(_) => return None,
+						};
+
+						// Build new task.
+						let ah_task = Scheduled {
+							maybe_id: rc_task.maybe_id,
+							priority: rc_task.priority,
+							call: ah_call,
+							maybe_periodic: rc_task.maybe_periodic,
+							origin: ah_origin,
+						};
+						ah_tasks.push(Some(ah_task));
+					}
+				}
+				// Filter out blocks that end up with no valid tasks after conversion.
+				if !ah_tasks.is_empty() {
+					Some((block_number, ah_tasks))
+				} else {
+					None
+				}
+			})
+			.collect();
+
+		// Collect agendas on AH.
+		let mut ah_agenda: Vec<_> = pallet_rc_migrator::scheduler::alias::Agenda::<T>::iter()
+			.map(|(bn, tasks)| (bn, tasks.into_inner()))
+			.collect();
+
+		// Assert storage 'Scheduler::Agenda::ah_post::length'
+		assert_eq!(
+			ah_agenda.len(),
+			expected_ah_agenda.len(),
```

Comment:
There is some oddity with the scheduler that it can also have an all empty agenda, so if it fails here then we know that it could be triggered by that.